### PR TITLE
Adds exotic and talking corgi crates

### PR DESCRIPTION
**Pull request by chesse20**
[Changelogs]: 

:cl: 
add: Adds Exotic Corgi Crate to Cargo
add: Adds Exotic Corgi, a Corgi with a random hue applied to it
add: Adds Talking Corgi Crate, and talking Corgi

/:cl:

[why]: Fun new additions to cargo. Can you get the extremely rare jet black corgi?

### CODE REVIEW

**Comment by lordsaladin** [comment]
Yes.

**Comment by Ratvar** [comment]
Is there a ghost corgi that passes through things and corgi without sprite?
Cockroarch-sized and megafauna-sized corgi?

**Comment by MrDoomBringer** [comment]
while i feel like this is a fun update, surely you could do more with it? Actual sprites instead of hue shifts (so people dont think the corgi ate a crayon or something), maybe special effects? (@Ratvar 's idea sounds pretty fun!)

**Comment by chesse20** [comment]
If someone wants to make resprites of corgis i'll add them as base sprites for the exotic corgis

**Comment by PKPenguin321** [comment]
talking corgi crate its actually a guy in a corgi costume

**Comment by Time-Green** [comment]
Could you instead greyscale corgi fur and make it an overlay? That way their color isn't on their eyes and skin

**Comment by chesse20** [comment]
@PKPenguin321 added it

File: code/modules/cargo/packs.dm [hunk 1]
```diff
@@ -1645,6 +1645,44 @@
...
+	D.name = "exotic corgi"
+	D.color = rgb(rand(0, 255), rand(0, 255), rand(0, 255))
+
+/mob/living/carbon/human/talking_corgi/handle_status_effects()
+	..()
+	if(prob(5))
+		say(pick("Woof.", "Bark.", "Arf.", "Woof woof.", "Give me a treat."))
+
+/datum/supply_pack/critter/corgi/talking
+	name = "Talking Corgi Crate"
```
[the remark this answers is not in the document]
Comment (PKPenguin321) [reply code/modules/cargo/packs.dm]: thats not what it is

File: code/modules/cargo/packs.dm [hunk 2]
```diff
@@ -1645,6 +1645,44 @@
...
+/datum/supply_pack/critter/corgi/exotic/generate()
+	. = ..()
+	var/mob/living/simple_animal/pet/dog/corgi/D = locate() in .
+	D.name = "exotic corgi"
+	D.color = rgb(rand(0, 255), rand(0, 255), rand(0, 255))
+
+/mob/living/carbon/human/talking_corgi/handle_status_effects()
+	..()
+	if(prob(5))
+		say(pick("Woof.", "Bark.", "Arf.", "Woof woof.", "Give me a treat."))
```
[the remark this answers is not in the document]
Comment (PKPenguin321) [reply code/modules/cargo/packs.dm]: its a corgi that talks

**Comment by Tlaltecuhtli** [comment]
give it a simple ai (look at slime code)

File: code/modules/cargo/packs.dm (hunk 2 above)
[the remark this answers is not in the document]
Comment (praisenarsie) [reply code/modules/cargo/packs.dm]: > what the fuck

**Comment by chesse20** [comment]
@Tlaltecuhtli What do you mean, and how would i do that on a human mob?

File: code/modules/cargo/packs.dm [hunk 3]
```diff
@@ -1645,6 +1645,44 @@
...
+	H.reagents.add_reagent("whiskey", 80)
+	H.put_in_hands(new /obj/item/reagent_containers/food/drinks/bottle/whiskey)
+	H.equip_to_appropriate_slot(new /obj/item/clothing/suit/hooded/ian_costume)
+	H.equip_to_appropriate_slot(new /obj/item/clothing/head/hooded/ian_hood)
+	H.hair_style = "Skinhead"
+	H.facial_hair_style = "Neckbeard"
+	H.gender = MALE
+	H.update_body()
+	H.update_hair()
+	H.name = "talking corgi"
```
Comment (cyclowns) [inline code/modules/cargo/packs.dm]: why not have all this information set when the talking corgi mob initializes, rather than when the crate does?

Comment (cyclowns) [reply code/modules/cargo/packs.dm]: also maybe think about making the corgi costume an actual coded outfit so you dont have to use equip_to_appropriate_slot

Comment (chesse20) [reply code/modules/cargo/packs.dm]: i didnt care

Comment (cyclowns) [reply code/modules/cargo/packs.dm]: what

File: code/modules/cargo/packs.dm (hunk 1 above)
[the remark this answers is not in the document]
Comment (KingPhilipIII) [reply code/modules/cargo/packs.dm]: Go back to Citidal Kevin.

**Review by SpaceManiac — CHANGES_REQUESTED** [review]
You don't have to bother with an outfit datum if you don't want but all these var changes should at the very least be moved to `talking_corgi/Initialize()` rather than tied to the supply crate.

**Comment by chesse20** [comment]
ill work on it

**Comment by chesse20** [comment]
@SpaceManiac made the changes

**Comment by optimumtact** [comment]
the talking corgi needs to be removed but the rest is fine

**Comment by fluffe9911** [comment]
Rip talking corgi you will be missed

**Comment by chesse20** [comment]
why

**Comment by optimumtact** [comment]
@ShizCalev you serious?

**Pull request merged**